PsfLauncher/main.cpp

```diff
@@ -13,223 +13,227 @@
-                if (((unsigned long long)(shex.hInstApp)) > 32)
-                    return 0;
-                else
-                    return (int)(unsigned long long)(shex.hInstApp);
-            case WAIT_FAILED:
-                return ::GetLastError();
-
-            default:
-                return ERROR_INVALID_HANDLE;
-            }			
+        else if (endingScriptError.IsThereAnError())
+        {
+            ::PSFReportError(endingScriptError.Print().c_str());
+            return endingScriptError.GetErrorNumber();
         }
     }
+
+    return 0;
 }
 catch (...)
 {
     ::PSFReportError(widen(message_from_caught_exception()).c_str());
     return win32_from_caught_exception();
+}
 
+ErrorInformation RunScript(const psf::json_object * scriptInformation, std::filesystem::path packageRoot, LPCWSTR dirStr, int cmdShow)
+{
+    //Script arguments are optional.
+    auto scriptArgumentsJObject = scriptInformation->try_get("scriptArguments");
+    std::wstring scriptArguments;
+    if (scriptArgumentsJObject)
+    {
+        scriptArguments = scriptArgumentsJObject->as_string().wide();
+    }
+
+    std::wstring scriptPath = scriptInformation->get("scriptPath").as_string().wide();
+    std::filesystem::path powershellScriptPath(scriptPath);
+
+    std::wstring powershellCommandString(L"powershell.exe -file ");
+
+    powershellCommandString.append(scriptPath);
+    powershellCommandString.append(L" ");
+    powershellCommandString.append(scriptArguments);
+
+    Log("Looking for the script here: ");
+    LogStringW("Script Path", scriptPath.c_str());
+    auto currentDirectory = (packageRoot / dirStr);
+    auto doesFileExist = std::filesystem::exists(currentDirectory / powershellScriptPath);
+
+    if (doesFileExist)
+    {
+        ExecutionInformation execInfo;
+        execInfo.ApplicationName = nullptr;
+        execInfo.CommandLine = powershellCommandString;
+        execInfo.CurrentDirectory = currentDirectory.c_str();
+
+        //RunInAppEnviorment is optional.
+        auto runInVirtualEnviormentJObject = scriptInformation->try_get("runInVirtualEnvironment");
+        auto runInVirtualEnviorment = false;
+
+        if (runInVirtualEnviormentJObject)
+        {
+            runInVirtualEnviorment = runInVirtualEnviormentJObject->as_string().wide();
+        }
+
+        return StartProcess(execInfo, cmdShow, runInVirtualEnviorment);
+    }
+    else
+    {
+        std::wstring errorMessage = L"The powershell file ";
+        errorMessage.append(currentDirectory / powershellScriptPath);
+        errorMessage.append(L" can't be found");
+        ErrorInformation error(errorMessage, ERROR_FILE_NOT_FOUND);
+
+        return error;
+    }
 }
 
+ErrorInformation GetAndLaunchMonitor(const psf::json_object * monitor, std::filesystem::path packageRoot, int cmdShow, LPCWSTR dirStr)
+{
+    bool asadmin = false;
+    bool wait = false;
+    auto monitor_executable = monitor->try_get("executable");

```

Comment:
casing is inconsistent - here it's snake_case, above it's camelCase

Reply:
Fixed.